Document use of ambient wrapper chart in 1.23 install/upgrade guides (#15279)
istio/istio.io: content/en/docs/ambient/install/helm-installation/index.md

*See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation.*

> {{< gloss "ambient" >}}Ambient{{< /gloss >}} is made up of multiple components - for production deployments, it is strongly recommended to install the components separately to allow for more control during upgrades. Consult the [operations guidelines](/docs/ops/deployment) for production deployment considerations.

Reviewer: ```suggestion
Istio comprises many components (control plane, data plane, custom resources, etc). For production deployments, it is strongly recommended to install the components separately to allow for fine grained control during upgrades. Consult the [operations guidelines](/docs/ops/deployment) for production deployment considerations.
```

so is sidecar mode.  At least three of the charts are shared?

Does the operations docs actually talk about what you should do differently?  Is it a scaling suggestion? (You wouldn't scale istio-cni and I doubt you'd scale ztunnel)

Author: > so is sidecar mode. At least three of the charts are shared?

Yeah, this is sort of boilerplate. I'll enhance to mention that some components have different impacts on app traffic and node health than others (data plane vs control plane, really) and this impacts upgrade cadence considerations. 

> Does the operations docs actually talk about what you should do differently? Is it a scaling suggestion? (You wouldn't scale istio-cni and I doubt you'd scale ztunnel)

Scaling, and upgrade cadence.

Really, this should link to

https://preliminary.istio.io/latest/docs/ops/best-practices/deployment/

and that should be built out. I'll do that as part of this.

Author: Actually - what I was kinda hoping is that we could just link to wherever this https://github.com/istio/istio.io/pull/15247 ends up (under `/ops/` or wherever) as the "recommended/opinionated production setup".

> If you just want to try out Istio ambient in a non-production cluster, to keep things simple it is recommended to use the `ambient` Helm wrapper chart which installs the minimum set of required components together.

Reviewer: ```suggestion
For installing Istio with support for the {{< gloss "ambient" >}}ambient data plane mode{{< /gloss >}} in a non-production cluster, we provide a simple Helm chart which bundles all the required components.
```

> ## Install the components together

Reviewer: ```suggestion
## Simple install
```

(or similar)

> The `ambient` Helm wrapper chart composes all the components that make up an ambient installation into one Helm chart:

Reviewer: is "wrapper chart" a term of art?  I don't see it in the Helm docs

Author: very much not.

This is a relatively common pattern (used in Argo, etc) that doesn't really have a specific Helm definition.

We could say "parent chart" or "composite chart" but to Helm it's just a Helm chart with Helm dependencies (and nothing else)

Author: I've just removed `wrapper` as a term, and talked around the informal term. Open to better ideas if anyone has them.

> The `ambient` Helm wrapper chart composes all the components that make up an ambient installation into one Helm chart:

Reviewer: ```suggestion
The `ambient` Helm chart composes all the components that support Istio with the ambient data plane:
```

## Uninstall

> If you installed with the `ambient` wrapper chart above, you can uninstall Istio and its components by uninstalling that chart:

Reviewer: ```suggestion
If you installed with the simple `ambient` chart above, you can uninstall Istio and its components by uninstalling that chart:
```

maybe don't put this before the gateway instructions?

Author: Yeah the idea was to exactly reverse the install directions but I dorked the wording here.

1. Delete any Istio gateway chart installations (optional, if installed):

    {{< text syntax=bash snip_id=delete_ingress >}}
    $ helm delete istio-ingress -n istio-ingress --wait
    $ kubectl delete namespace istio-ingress
    {{< /text >}}

### Uninstall the components together

1. Delete the Istio ambient chart:

    {{< tip >}}
    [By design](https://github.com/helm/community/blob/main/hips/hip-0011.md#deleting-crds),
    deleting a chart via Helm doesn't delete the installed Custom
    Resource Definitions (CRDs) installed via the chart.
    {{< /tip >}}

    {{< text syntax=bash snip_id=delete_ambient_wrapper >}}
    $ helm delete istio-ambient -n istio-system --wait
    {{< /text >}}

> 1. Delete the `istio-system` namespace:

Reviewer: this isn't done automatically? 🤔

Author: It would be, except for the whole "you have to delete CRDs manually" thing, which Helm categorically refuses to do automatically, because kubernetes effectively provides no way to do it safely and unconditionally
